pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java

```diff
@@ -80,4 +85,87 @@
   double[] getDoubleArray(int rowId, int colId);
 
   String[] getStringArray(int rowId, int colId);
+
+  /* The MetadataKeys is used in V3, where we present metadata as Map<MetadataKeys, String>
+   * ATTENTION:
+   *  - Don't change existing keys.
+   *  - Don't remove existing keys.
+   *  - Always add new keys to the end.
+   *  Otherwise, backward compatibility will be broken.
+   */
+  enum MetadataKeys {
```

Comment:
I still suggest associating an id with each key instead of using ordinal of the enum. The convention here should be always increasing the id when adding new keys.
Id is more flexible than ordinal for the following reasons:
- Ordinal works as always putting the index key as the id. If by any chance people accidentally change the order of the keys, it will break
- With id, we can remove keys in a backward-compatible way in two releases if necessary. With ordinal, we have to keep a place holder so that the ordinal for other keys don't change

@mqliang @siddharthteotia @mcvsubbu Thoughts?

Reply:
@Jackie-Jiang , I prefer enums. We can add a unit test that asserts (A < B< C ...), to catch any re-orders.
If we have to manually insert a value, then duplicate values are possible (by mistake) and that can also cause problems.

Reply:
How about let's use enum at this moment. We can discuss more, if we decide to associate an id with each key later on, as long as we associate the first key with 0, second with 1, third key with 3...The bytes send on wire will not change, we can address it in a separate PR, it's just some code level change, will not change any payloads send on wire.

Reply:
Yes, we can argue both ways here but my preference would be enum with implicit ordinal as opposed to id based. I agree the latter gives more flexibility to the user but I don't think we need it. So a simple enum with ordinal as id along with clear javadoc highlighting the rules for updating the enum is preferable imo.